keras_retinanet/preprocessing/csv_generator.py

```diff
@@ -169,14 +169,35 @@ def num_classes(self):
         """
         return max(self.classes.values()) + 1
 
+    def has_label(self, label):
+        """ Return True if label is a known label.
+        """
+        return label in self.labels
+
+    def has_name(self, name):
+        """ Returns True if name is a known class.
+        """
+        return name in self.classes
+
     def name_to_label(self, name):
         """ Map name to label.
+
+        Returns
+            The label of name. Returns -1 if it doesn't exist.
         """
+        if name not in self.classes:
+            raise ValueError("Invalid class name ({}) provided, options are {}.".format(name, self.classes.keys()))
```

Comment:
Should probably be a `KeyError`.

Also, it is probably better in python to just attempt the lookup and catch the `KeyError` and then throw a prettier key error.